ntoskrnl/ke/amd64/ipi.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     IPI code for x64
+ * COPYRIGHT:   Copyright 2023 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* FUNCTIONS *****************************************************************/
+
+VOID
+FASTCALL
+KiIpiSend(
+    _In_ KAFFINITY TargetSet,
+    _In_ ULONG IpiRequest)
+{
+    /* Check if we can send the IPI directly */
+    if (IpiRequest == IPI_APC)
+    {
+        HalSendSoftwareInterrupt(TargetSet, APC_LEVEL);
+    }
+    else if (IpiRequest == IPI_DPC)
+    {
+        HalSendSoftwareInterrupt(TargetSet, DISPATCH_LEVEL);
+    }
+    else if (IpiRequest == IPI_FREEZE)
+    {
+        HalSendNMI(TargetSet);
```

Comment:
I would suggest adding a small comment explaining why the IPI_FREEZE would map to an NMI here.

Reply:
[A bit like this](https://preview.redd.it/iuy9fxt300811.png?auto=webp&s=156f5ffebbea5b30189eb36012e5bc7e3369c381)
😄